plugin/core/edit.py

```diff
@@ -13,19 +14,22 @@ def parse_workspace_edit(workspace_edit: Dict[str, Any]) -> Dict[str, List[TextE
     if 'documentChanges' in workspace_edit:
         for document_change in workspace_edit.get('documentChanges', []):
             uri = document_change.get('textDocument').get('uri')
-            changes[uri_to_filename(uri)] = list(parse_text_edit(change) for change in document_change.get('edits'))
+            version = document_change.get('textDocument').get('version')
```

Comment:
While you are at it, can you add `workspace.workspaceEdit.documentChanges` to the `session.py capabilities`
[Workspace Edit](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspaceEdit)
```
export interface WorkspaceEdit {
	 ...

	/**
	 * Depending on the client capability `workspace.workspaceEdit.resourceOperations` document changes
	 * are either an array of `TextDocumentEdit`s to express changes to n different text documents
	 * where each text document edit addresses a specific version of a text document. Or it can contain
	 * above `TextDocumentEdit`s mixed with create, rename and delete file / folder operations.
	 *
	 * Whether a client supports versioned document edits is expressed via
	 * `workspace.workspaceEdit.documentChanges` client capability.
	 *
	 * If a client neither supports `documentChanges` nor `workspace.workspaceEdit.resourceOperations` then
	 * only plain `TextEdit`s using the `changes` property are supported.
	 */
	documentChanges?: (TextDocumentEdit[] | (TextDocumentEdit | CreateFile | RenameFile | DeleteFile)[]);
}

```

Reply:
Done. Thanks.

Reply:
And one small note, if we ever in the future add `resourceOperations`,
that means that potentially this code will be partially broken. 

Because then, `document_change` is not guaranteed to be an instance of `TextDocumentEdit`, it can also be  `CreateFile | RenameFile | DeleteFile`.

These types don't have a `textDocument` on them, so it will throw an error when trying to get the version. `.get('version')`

So I think it is better to worry about those things right now than later. :)

Reply:
Small typo in 45c4096dff87632f635e7d8a619eaa5c2b2f4a37 :D 
instead of: 
```
   "workspace": {
            ...
            "documentChanges: True
```
should be:
```
"workspace" {
  "workspaceEdit: {
    "documentChanges": True
--

Reply:
Ops